### PR TITLE
Add Lazy configuration loader

### PR DESCRIPTION
**Pull request by JoeAlisson**

### CODE REVIEW

**Comment by LeonanCarvalho** [comment]
The semantic isn't bad. But it made a lot of changes and I think @Tayran  is working at config engine.

The Pascal is good for understanding but not so easy to make a "configurator" or a similar program to help configure values, the model using dots we can use a `split` function to create a sort of aggregation, but I'm not sure if its applicable for now.

I agree for reverting for now to analyze the Lazy loading configuration.

**Comment by Elfocrash** [comment]
The semantic is terrible the way it is implemented. dots are used to separate context.
They are not used to take the place of spaces.

Please revert it because from what i've seen the rest of the implementation is decent.

**Comment by JoeAlisson** [comment]
There is no problem @Elfocrash. At night I'll do that

**Comment by Elfocrash** [comment]
@JoeAlisson Awesome, thank you

File: L2JBrasil_CORE/java/com/it/br/util/Util.java [hunk 1]
```diff
@@ -42,6 +43,10 @@
 
 public class Util
 {
+	/**
+	 * Comment for <code>_1000</code>
+	 */
+	private static final int SECOND = 1000;
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/util/Util.java]: ONE_SECOND_IN_MILLISECONDS

File: L2JBrasil_CORE/java/com/it/br/util/Util.java [hunk 2]
```diff
@@ -241,5 +246,18 @@ public static void printGeneralSystemInfo()
...
 		printJvmInfo();
 	}
-
+	
+	
+	public static boolean isEmpty(String string) {
+		return string == null || string.isEmpty();
+	}
+	
+	public static boolean isEmpty(Collection<?> collection) {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/util/Util.java]: Rename to isNullOrEmpty

File: L2JBrasil_CORE/java/com/it/br/util/Util.java [hunk 3]
```diff
@@ -241,5 +246,18 @@ public static void printGeneralSystemInfo()
...
+		return collection == null || collection.isEmpty();
+	}
+	
+	public static int getInSeconds(int val) {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/util/Util.java]: Get what in seconds? Days in seconds? Hours in seconds? Give appropriate name.

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	
+	public int getTimeBlockAfterBan() {
+		return timeBlockAfterBan;
+	}
+
+	public boolean isEnableAnyServer() {
+		return enableAnyServer;
+	}
+	
+	public boolean isShowLicence() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java]: isShowLicence is not readable. Change to showLicense

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+		return ShowLicence;
+	}
+	
+	public boolean isEnableAutoCreateAccount() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java]: isEnableAutoCreateAccount is not english. Change to autoCreateAccountsEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	
+	public int getTriesBeforeBan() {
+		return triesBeforeBan;
+	}
+	
+	public int getTimeBlockAfterBan() {
+		return timeBlockAfterBan;
+	}
+
+	public boolean isEnableAnyServer() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java]: This doesn't mean anything to the reader. Change please

**Comment by Elfocrash** [comment]
@JoeAlisson Sorry but i said Pascal case.... 😢 
If it's too much to do don't worry about it.

File: L2JBrasil_CORE/java/com/it/br/configuration/Configurator.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
+package com.it.br.configuration;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import com.it.br.configuration.settings.Settings;
+
+/* This program is free software; you can redistribute it and/or modify
```
[the remark this answers is not in the document]
Comment (JoeAlisson) [reply L2JBrasil_CORE/java/com/it/br/configuration/Configurator.java]: You need to update the project profile. It still use the v2.

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public boolean isEnabledRewriteBatchedStatement() {
+		return enabledRewriteBatchedStatement;
+	}
+	
+	public boolean isEnabledMaintainTimestats() {
+		return enabledMaintainTimestats;
+	}
+	
+	public boolean isEnabledAutoCommit() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: autoCommitEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public boolean isEnabledLocalTransactionState() {
+		return enableLocalTransactionState;
+	}
+	
+	public boolean isEnabledRewriteBatchedStatement() {
+		return enabledRewriteBatchedStatement;
+	}
+	
+	public boolean isEnabledMaintainTimestats() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: maintainTimestatsEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public int getPreparedStatementCacheSize() {
+		return PreparedStatementCacheSize;
+	}
+	
+	public int getLimitSqlCache() {
+		return limitSqlCache;
+	}
+	
+	public boolean isEnabledResultsetCache() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: resultsetCacheEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+		return enabledResultsetCache;
+	}
+	
+	public boolean isEnabledPreparedStatementCache() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: preparetedStatementCacheEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+		return enabledPreparedStatementCache;
+	}
+	
+	public boolean isEnabledServerPreparedStatement() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: serverPreparedStatementEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public boolean isEnabledLocalSessionState() {
+		return enabledLocalSessionState;
+	}
+	
+	public boolean isEnabledLocalTransactionState() {
+		return enableLocalTransactionState;
+	}
+	
+	public boolean isEnabledRewriteBatchedStatement() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: rewriteBatchedStatementEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public boolean isEnabledServerPreparedStatement() {
+		return enabledServerPreparedStatement;
+	}
+	
+	public boolean isEnabledLocalSessionState() {
+		return enabledLocalSessionState;
+	}
+	
+	public boolean isEnabledLocalTransactionState() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: localTransactionStateEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public boolean isEnabledPreparedStatementCache() {
+		return enabledPreparedStatementCache;
+	}
+	
+	public boolean isEnabledServerPreparedStatement() {
+		return enabledServerPreparedStatement;
+	}
+	
+	public boolean isEnabledLocalSessionState() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: localSessionStateEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+	
+	public void setMaxConnections(int count) {
+		this.maxConnections = count;
+	}
+	
+	public int getConnectionIdleTimeout() {
+		return connectionIdleTimeout;
+	}
+	
+	public boolean isEnabledServerCache() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/DatabaseSettings.java]: serverCacheEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+	
+	public int getTriesBeforeBan() {
+		return triesBeforeBan;
+	}
+	
+	public int getTimeBlockAfterBan() {
+		return timeBlockAfterBan;
+	}
+
+	public boolean isEnabledAnyServer() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java]: anyServerEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+		return enabledAnyServer;
+	}
+	
+	public boolean isEnabledShowLicence() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java]: showLicence

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+		return enabledShowLicence;
+	}
+	
+	public boolean isEnabledAutoCreateAccount() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/LoginSettings.java]: autoCreateAccountsEnabled

File: L2JBrasil_CORE/java/com/it/br/configuration/settings/NetworkSettings.java [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+	public String getLoginExternalHostname() {
+		return loginExternalHostname;
+	}
+	
+	public String getLoginInternalHostname() {
+		return loginInternalHostname;
+	}
+	
+
+	public boolean isEnabledTelnet() {
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/configuration/settings/NetworkSettings.java]: telnetEnabled

File: L2JBrasil_CORE/java/com/it/br/loginserver/LoginController.java [hunk 21]
```diff
@@ -19,23 +12,41 @@
...
+import com.it.br.util.Rnd;
+
 /**
  * This class ...
  *
  * @version $Revision: 1.7.4.3 $ $Date: 2005/03/27 15:30:09 $
  */
 public class LoginController
 {
+	private static final int SECOND = 1000;
```
Comment (Elfocrash) [inline L2JBrasil_CORE/java/com/it/br/loginserver/LoginController.java]: ONE_SECOND_IN_MILLISECONDS

**Comment by Elfocrash** [comment]
You missed two changes but it's fine. 
Have you tested it?
cc @Tayran @LeonanCarvalho

**Comment by Elfocrash** [comment]
Cool, as long as it is tested, It's fine by me.

@Tayran @LeonanCarvalho To whoever merges this, DO NOT MERGE PULL REQUEST. Click the arrow and select SQUASH AND MERGE. Let's keep the logs clean.

**Pull request merged**